pkg/storage/stores/indexshipper/uploads/table.go

```diff
@@ -0,0 +1,139 @@
+package uploads
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+
+	"github.com/grafana/loki/pkg/storage/stores/indexshipper/index"
+	"github.com/grafana/loki/pkg/storage/stores/shipper/storage"
+	util_log "github.com/grafana/loki/pkg/util/log"
+)
+
+const (
+	// a temp file is created during uploads with name of the db + tempFileSuffix
+	tempFileSuffix = ".temp"
+)
+
+type Table interface {
+	Name() string
+	AddIndex(userID string, idx index.Index) error
+	ForEach(userID string, callback index.ForEachIndexCallback) error
+	Upload(ctx context.Context) error
+	Cleanup(indexRetainPeriod time.Duration) error
+	Stop()
+}
+
+// table is a collection of one or more index files created by the ingester or compacted by the compactor.
+// A table would provide a logical grouping for index by a period. This period is controlled by `schema_config.configs.index.period` config.
+// It contains index for all the users sending logs to Loki.
```

Comment:
is this still correct? I thought I heard we were using per tenant indices now?

Reply:
yeah, a table still contains index for all the users. We just build an index file per tenant which is still stored within a table.